src/providers/v2/quote-provider.ts

```diff
@@ -157,54 +103,11 @@ export class V2QuoteProvider implements IV2QuoteProvider {
           } else {
             let inputAmount = amount.wrapped;
 
-            for (let i = route.pairs.length - 1; i >= 0; i--) {
-              const pair = route.pairs[i]!;
-              if (amount.wrapped.currency.buyFeeBps) {
-                // this should never happen, but just in case it happens,
-                // there is a bug in sor. We need to log this and investigate.
-                const error =
-                  new Error(`Buy fee bps should not exist on input amount
-                ${JSON.stringify(amount)} on amounts ${JSON.stringify(amounts)}
-                on routes ${JSON.stringify(routes)}`);
-
-                // artificially create error object and pass in log.error so that
-                // it also log the stack trace
-                log.error(
-                  { error },
-                  'Buy fee bps should not exist on input amount'
-                );
-                metric.putMetric(
-                  'V2_QUOTE_PROVIDER_INCONSISTENT_BUY_FEE_BPS_VS_FEATURE_FLAG',
-                  1,
-                  MetricLoggerUnit.Count
-                );
-              }
-
-              if (providerConfig.enableFeeOnTransferFeeFetching) {
-                if (
-                  pair.token0.equals(inputAmount.currency) &&
-                  pair.token0.buyFeeBps?.gt(BigNumber.from(0))
-                ) {
-                  const inputAmountWithBuyFeeBps = CurrencyAmount.fromRawAmount(
-                    pair.token0,
-                    inputAmount.quotient
-                  );
-                  [inputAmount] = pair.getInputAmount(inputAmountWithBuyFeeBps);
-                } else if (
-                  pair.token1.equals(inputAmount.currency) &&
-                  pair.token1.buyFeeBps?.gt(BigNumber.from(0))
-                ) {
-                  const inputAmountWithBuyFeeBps = CurrencyAmount.fromRawAmount(
-                    pair.token1,
-                    inputAmount.quotient
-                  );
-                  [inputAmount] = pair.getInputAmount(inputAmountWithBuyFeeBps);
-                } else {
-                  [inputAmount] = pair.getInputAmount(inputAmount);
-                }
-              } else {
-                [inputAmount] = pair.getInputAmount(inputAmount);
-              }
+            for (const pair of route.pairs.reverse()) {
```

Comment:
Please help me double check that this for loop remains the same

Reply:
Looks fine!